### PR TITLE
Fix issues related to logging, refactor WindDial component

### PR DESCRIPTION
**Pull request by pjanik**
[#176563064]

This PR fixes some issues related to logging. Mostly it prevents components from generating thousands of events by using `onChangeCommitted` / `onChangeEnd` handlers instead of `onChange`. I've also cleaned up `WindDial` component a bit.

### CODE REVIEW

**Review by knowuh — APPROVED** [review]
Thank you this looks great.

Small comments for discussion only.

File: src/components/terrain-panel.tsx [hunk 1]
```diff
@@ -187,13 +189,27 @@ export class TerrainPanel extends BaseComponent<IProps, IState> {
...
-    simulation.updateZoneVegetation(this.selectedZone, newVegetationType);
+    if (zone.vegetation !== newVegetationType) {
+      simulation.updateZoneVegetation(this.selectedZone, newVegetationType);
+    }
+  }
+
+  public handleVegetationChangeCommitted = (event: React.ChangeEvent<HTMLInputElement>, value: number) => {
+    const { simulation } = this.stores;
+    const zone = simulation.zones[this.selectedZone];
+    const newVegetationType = zone.terrainType === TerrainType.Mountains ? value + 1 : value;
```
[the remark this answers is not in the document]
Comment (pjanik) [reply src/components/terrain-panel.tsx:200]: Honestly, I agree it's terrible, but I didn't try to refactor too much of the code in this dev cycle. I think the slider should just have a different range and we shouldn't have to check that in the handler, not sure why it got implemented that way. I'll try to fix that.

[the remark this answers is not in the document]
Comment (pjanik) [reply src/components/terrain-panel.tsx]: I agree, I don't know why it got implemented that way. I've fixed that here:
https://github.com/concord-consortium/wildfire-model/pull/61/commits/79c4771719a39affc39120dbffa35ed1be1dcfb3
I just moved this logic to slider itself, as I think its internals and event handlers / parent component shouldn't care.

**Comment by pjanik** [comment]
Thanks, I've addressed the comment, so I'm going to merge this PR.

**Pull request merged**